core/main/src/service/telemetry_builder.rs

```diff
@@ -199,4 +203,44 @@ impl TelemetryBuilder {
             error!("send_telemetry={:?}", e)
         }
     }
+
+    pub fn start_firebolt_metrics_timer(
+        extn_client: &ExtnClient,
+        name: String,
+        app_id: String,
+    ) -> Option<Timer> {
+        let metrics_context = extn_client.get_metrics_context();
+
+        if !metrics_context.enabled {
+            return None;
+        }
+
+        let metrics_tags = get_metrics_tags(extn_client, InteractionType::Firebolt, Some(app_id));
+
+        debug!("start_firebolt_metrics_timer: {}: {:?}", name, metrics_tags);
+
+        Some(Timer::start(
+            name,
+            metrics_context.device_session_id.clone(),
+            Some(metrics_tags),
+        ))
+    }
+
+    pub fn stop_and_send_firebolt_metrics_timer(
+        ps: &PlatformState,
+        timer: Option<Timer>,
+        status: String,
+    ) {
+        if let Some(mut timer) = timer {
+            timer.stop();
+            timer.insert_tag(Tag::Status.key(), status);
+            debug!("stop_and_send_firebolt_metrics_timer: {:?}", timer);
+            if let Err(e) = Self::send_telemetry(ps, TelemetryPayload::Timer(timer)) {
+                error!(
+                    "stop_and_send_firebolt_metrics_timer: send_telemetry={:?}",
+                    e
+                )
+            }
+        }
+    }
```

Comment:
These are utility functions to minimize boilerplate throughout anything being instrumented. There are similar functions specific to services rather than firebolt.